### PR TITLE
Add utils for retrieving backendconfigs for service (and reversely)

### PR DESCRIPTION
**Pull request by MrHohn**
/assign @rramkumar1 @nicksardo 
cc @bowei

### CODE REVIEW

File: pkg/backendconfig/backendconfig.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+
+	"k8s.io/ingress-gce/pkg/annotations"
+	backendconfigv1alpha1 "k8s.io/ingress-gce/pkg/apis/backendconfig/v1alpha1"
+)
+
+// GetServicesForBackendConfig returns all services that reference the given
+// BackendConfig.
+func GetServicesForBackendConfig(svcLister cache.Store, backendConfig *backendconfigv1alpha1.BackendConfig) []*apiv1.Service {
```
Comment (MrHohn) [inline pkg/backendconfig/backendconfig.go]: Planing to plug this to trigger Ingress sync.

File: pkg/backendconfig/backendconfig.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+				}
+			}
+		}
+	}
+	return svcs
+}
+
+// GetBackendConfigForServicePort returns the corresponding BackendConfig for
+// the given ServicePort if specified.
+func GetBackendConfigForServicePort(backendConfigLister cache.Store, svc *apiv1.Service, svcPort *apiv1.ServicePort) (*backendconfigv1alpha1.BackendConfig, error) {
```
Comment (MrHohn) [inline pkg/backendconfig/backendconfig.go]: Planing to plug this into `servicePort()`:
https://github.com/kubernetes/ingress-gce/blob/16d4f399d64f6c0dfafe54dc73356cee2d13f50c/pkg/backends/utils.go#L50-L95.

**Comment by rramkumar1** [comment]
/lgtm
Leaving to nicksardo@ for approval.

**Comment by MrHohn** [comment]
Thanks for the review!

**Pull request merged**